src/partition.rs

```diff
@@ -0,0 +1,39 @@
+use fluvio::metadata::partition::PartitionSpec;
+use wasm_bindgen::prelude::*;
+
+use fluvio::metadata::objects::Metadata;
+
+#[wasm_bindgen]
+pub struct PartitionMetadata {
+    inner: Metadata<PartitionSpec>,
+}
+
+#[wasm_bindgen]
+impl PartitionMetadata {
+    #[wasm_bindgen(method, getter)]
+    pub fn name(&self) -> String {
+        self.inner.name.clone()
+    }
+
+
+    #[wasm_bindgen(method, getter = highWatermark)]
+    pub fn high_watermark(&self) -> u32 {
+        self.inner.status.leader.hw as u32
+    }
+
+    #[wasm_bindgen(method, getter = logEndOffset)]
+    pub fn log_end_offset(&self) -> u32 {
+        self.inner.status.leader.leo as u32
+    }
+
+    #[wasm_bindgen(method, getter = liveReplicas)]
+    pub fn live_replicas(&self) -> u32 {
+        self.inner.status.lsr
```

Comment:
This looks like a typo in fluvio, right?

Reply:
I'm not sure what the `lsr` stuff is. But it looks like you could use [`self.inner.status.lsr()`](https://github.com/infinyon/fluvio/blob/eff4820d437f0961352c4a287c6b93c89517993f/src/controlplane-metadata/src/partition/status.rs#L86-L88)

Reply:
Sure, according to [this](https://github.com/infinyon/fluvio/blob/e72b67bac332d4060e2eb01de3923094a0b3db71/k8-util/helm/fluvio-sys/templates/crd_partition.yaml#L57) it means Live Replicas